lib/rubygems/ext/builder.rb

```diff
@@ -73,11 +73,13 @@ def self.run(command, results, command_name = nil)
       results << (command.respond_to?(:shelljoin) ? command.shelljoin : command)
 
       redirections = verbose ? {} : {err: [:child, :out]}
-      IO.popen(command, "r", redirections) do |io|
-        if verbose
-          IO.copy_stream(io, $stdout)
-        else
-          results << io.read
+      Bundler.with_clean_env do
```

Comment:
Can you add the explicitly loading `bundler`? We only activate it with https://github.com/rubygems/rubygems/blob/master/lib/rubygems.rb#L1191